src/crimson/os/seastore/cache.h

```diff
@@ -414,19 +415,18 @@ class Cache {
         SUBDEBUGT(seastore_cache,
             "{} {}~0x{:x} is present on t without fully loaded, reading ... -- {}",
             t, T::TYPE, offset, length, *ret);
-        return do_read_extent_maybe_partial<T>(ret->cast<T>(), 0, length);
+        return do_read_extent_maybe_partial<T>(ret->cast<T>(), 0, length, &t_src);
```

Comment:
Should this be `offset~length` instead of `0~length`?

Reply:
I think not, `offset` here is the extent physical address, but `do_read_extent_maybe_partial` expects an extent offset to read (>0 only if partially loaded).

Here is a full read request, so this is `0~length`.

Reply:
Oh, BTW, it seems that there's no user of `Cache::get_caching_extent()`, should we remove it?

Reply:
`test_seastore_cache.cc` is still using it, probably good to help validate whether the cache index is consistent with lba parent-child pointers pinning the cache.

Probably we can remove it upon dropping the cache index.